### PR TITLE
Implement Wellington design

### PR DESCRIPTION
**Pull request by PeeterPaal**
Implements Wellington design, which is a minimal base layout for developing new sites. The layout includes most used components, a singular layout (Front page), and minimal JS/CSS files.

Compared to previous Voog designs, Bower and Grunt are not used. Instead, building and minifying JS/CSS/assets is done using Taskfile. All required packages will be installed via NPM.

For testing and initializing dev environment:

Firstly, clone repo and install packages by running `npm install`

Try different tasks provided in the Taskfile. These include:
- `npm run dev` – initializes watchers for JS/CSS/assets. Process these files automatically whenever something changes.
- `npm run dev:kit` – does the same as previous, but also initializes Voog Kit to automatically push changes to site defined in `.voog` config file.
- `npm run prod` – builds and minifies JS/CSS/assets

Check that content areas work without issues and layout is usable:
- navigation menu is present
- language menu is present
- buttons are _button-like_ (paddings and borders are present)

### CODE REVIEW

File: components/html-head.tpl [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  media="screen"
+  rel="stylesheet"
+  type="text/css"
+/>
+
+{% if editmode %}
+  <link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-tools/latest/edicy-tools.css">
+{% endif %}
+
+<link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.css">
```
Comment (tanelj) [inline components/html-head.tpl]:
```suggestion
{% if site.search.enabled -%}
  {% stylesheet_link "libs/edicy-search/latest/edicy-search.css" static_host="true" %}
{%- endif %}
```

File: components/html-head.tpl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+<link
+  href="{{ stylesheets_path }}/main.min.css?v={{ template_settings.version }}"
+  media="screen"
+  rel="stylesheet"
+  type="text/css"
+/>
+
+{% if editmode %}
+  <link rel="stylesheet" href="{{ site.static_asset_host }}/libs/edicy-tools/latest/edicy-tools.css">
+{% endif %}
```
Comment (tanelj) [inline components/html-head.tpl]: Let's use [stylesheet_link](https://www.voog.com/developers/markup/tags/stylesheet-link) tag.
```suggestion
{% if editmode -%}
  {% stylesheet_link "libs/edicy-tools/latest/edicy-tools.css" static_host="true" %}
{%- endif %}
```

File: components/search.tpl [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<div class="search js-search js-modal-overlay">
+  <div class="search-inner js-modal">
+    <form id="search" class="search-form js-search-form" method="get" action="#">
+      <div class="search-left">
+        <input id="onpage_search" class="search-input js-search-input js-input-dynamic" type="text" name="" placeholder="{{  'search' | lc }}">
```
Comment (tanelj) [inline components/search.tpl]: Let's add [escape_once](https://www.voog.com/developers/markup/basics/filters#escape_once) filter to ensure that external string doesn't brake html.
```suggestion
        <input id="onpage_search" class="search-input js-search-input js-input-dynamic" type="text" name="" placeholder="{{ "search" | lc | escape_once }}">
```

File: components/search.tpl [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+      </div>
+      <div class="search-right">
+        <input class="search-submit js-search-submit" type="submit" value="Search" name="">
```
Comment (tanelj) [inline components/search.tpl]:
```suggestion
        <input class="search-submit js-search-submit" type="submit" value="{{ "search" | lc | escape_once }}" name="">
```

File: components/template-meta.tpl [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+{%- if article -%}
+  {%- assign og_obj = article -%}
+{%- elsif product -%}
+  {%- assign og_obj = product -%}
+{%- else -%}
+  {%- assign og_obj = page -%}
+{%- endif -%}
+
+<meta property="og:type" content="{%- if article -%}article{%- else -%}website{%- endif -%}">
+<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first:"/" }}">
```
Comment (tanelj) [inline components/template-meta.tpl]:
```suggestion
<meta property="og:url" content="{{ site.url }}{{ og_obj.url | remove_first: "/" }}">
```

File: components/template-meta.tpl [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+<meta property="og:title" content="{%- title -%}">
+<meta property="og:site_name" content="{{ page.site_title | escape }}">
```
Comment (tanelj) [inline components/template-meta.tpl]: [escape_once](https://www.voog.com/developers/markup/basics/filters#escape_once)` should be better.
```suggestion
<meta property="og:site_name" content="{{ page.site_title | escape_once }}">
```

File: components/javascripts.tpl [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{% comment %}SITE WIDE JAVASCRIPTS{% endcomment %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
+<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
+
+{% sitejs_include %}
+
+{% comment %}Site search related javascript components.{% endcomment %}
+{% if site.search.enabled %}
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
{% if site.search.enabled -%}
```

File: components/javascripts.tpl [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+  <script src="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.js"></script>
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', '{{ 'search_noresults' | lc | escape }}');</script>
+{% endif %}
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
{%- endif %}
```

File: components/javascripts.tpl [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{% comment %}SITE WIDE JAVASCRIPTS{% endcomment %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
+<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
+
+{% sitejs_include %}
+
+{% comment %}Site search related javascript components.{% endcomment %}
+{% if site.search.enabled %}
+  <script src="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.js"></script>
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', '{{ 'search_noresults' | lc | escape }}');</script>
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', {{ 'search_noresults' | lc | json }});</script>
```

File: components/javascripts.tpl [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+{% endif %}
+
+{% comment %}GOOGLE ANALYTICS INITIATION{% endcomment %}
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
{% comment %}SITE ANALYTICS INITIATION{% endcomment %}
```

File: components/menu-main.tpl [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    {% for item in site.visible_menuitems %}
+      {% menulink item wrapper-tag="li" %}
+      {% if item.current? and item.children? %}
+        <ul>
+          {% for child in item.children %}
+            {% menulink child wrapper-tag="li" %}
+          {% endfor %}
+
+          {% if editmode %}
+            <li>{% menuadd parent=item %}</li>
```
Comment (tanelj) [inline components/menu-main.tpl:22]: Second level "Add subpage"  button is not shown where the parent page is active, but has no children. It must be always shown for non root page.

File: components/menu-main.tpl [hunk 12]
```diff
@@ -0,0 +1,31 @@
+<nav class="menu-main">
+  <ul class="menu">
```
Comment (tanelj) [inline components/menu-main.tpl:2]: White spaces removal can be improved.

File: components/menu-lang.tpl [hunk 13]
```diff
@@ -0,0 +1,13 @@
+<div class="menu-lang">
+  {% for language in site.languages %}
```
Comment (tanelj) [inline components/menu-lang.tpl]:
```suggestion
  {% for language in site.languages -%}
```

File: components/menu-lang.tpl [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+    <span>
+      <a class="lang-flag lang-flag-{{ language.code }}{% if language.selected? %} active{% endif %}" href="{{ language.url }}" data-lang-code="{{ language.locale }}">{{ language.title }}</a>
+    </span>
+  {% endfor %}
```
Comment (tanelj) [inline components/menu-lang.tpl]:
```suggestion
  {%- endfor %}
```

File: components/menu-lang.tpl [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+
+  {% if editmode %}
```
Comment (tanelj) [inline components/menu-lang.tpl]:
```suggestion
  {% if editmode -%}
```

File: components/menu-lang.tpl [hunk 16]
```diff
@@ -0,0 +1,13 @@
...
+    <span class="edit-btn">
+      {% languageadd %}
+    </span>
+  {% endif %}
```
Comment (tanelj) [inline components/menu-lang.tpl]:
```suggestion
  {%- endif %}
```

File: components/javascripts.tpl [hunk 17]
```diff
@@ -0,0 +1,14 @@
+{% comment %}SITE WIDE JAVASCRIPTS{% endcomment %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
+<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
```
Comment (tanelj) [inline components/javascripts.tpl:3]:
```suggestion
<script src="{{ javascripts_path }}/application.min.js?v={{ template_settings.version }}"></script>
{% if editmode -%}
  <script src="{{ javascripts_path }}/editmode.min.js?v={{ template_settings.version }}"></script>
{%- endif %}
```

File: sources/stylesheets/_content-formatted.scss [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+
+  // Content area image title
+  .edy-image-container-with-title:after {
+    display: block;
+    padding: 4px;
+    font-size: 12px;
+    line-height: 1.3em;
+    content: attr(data-title);
+  }
+}
```
Comment (tanelj) [inline sources/stylesheets/_content-formatted.scss:132]: Please add image/map/video container related base styles: https://www.voog.com/developers/markup/basics/smarter-media

Please add form area basic CSS boilerplate (small/medium/large).

**Review by tanelj — CHANGES_REQUESTED** [review]
Nice work!

Here is first feedback mostly about Liquid tweaks.

File: javascripts/editmode.js [hunk 19]
```diff
@@ -0,0 +1,383 @@
...
+          target_width: 2048,
+          placeholder: '<div class="edy-img-drop-area-placeholder">' + placeholderText + '</div>',
+          removeBtn: null,
+          change: function (image) {
+            var $cropToggleButton = $contentItemBox.find('.js-toggle-crop-state');
+            $imgDropAreaTarget
+              .removeClass('is-cropped')
+              .addClass('not-cropped')
+              .css('opacity', .1)
+              ;
```
Comment (t-p-developer) [inline javascripts/editmode.js]: nit! Let's put `;` in the above line. Please check the rest of the file also, there is the same inconsistency.

File: javascripts/editmode.js [hunk 20]
```diff
@@ -0,0 +1,383 @@
...
+                    itemData.remove(cropStateKey, {
+                      success: function (data) {
+                        if (itemType !== 'article') {
+                          handleProductImage(placeholderText, itemId, null, $el);
+                        } else {
+                          addProductImagePlaceholder($el, placeholderText);
+                        }
+                      }
+                    });
+                  }
```
Comment (t-p-developer) [inline javascripts/editmode.js]: nit! Can we extract this logic? At the moment, it seems like a `callback-hell`...

File: javascripts/editmode.js [hunk 21]
```diff
@@ -0,0 +1,383 @@
...
+          }
+          var dataKey = e.target.dataset.name;
+          var dataId = e.target.dataset.id;
+          var dataEntity = e.target.dataset.entity;
+          var dataReload = e.target.dataset.reload;
+
+          if (dataEntity == 'site') {
+            var entityData = new Edicy.CustomData({
+              type: 'site'
+            });        } else {
```
Comment (t-p-developer) [inline javascripts/editmode.js]: nit! Formatting.

File: javascripts/editmode.js (hunk 20 above)
Comment (PeeterPaal) [reply javascripts/editmode.js]: This logic is copied over from other standard layouts. I would keep it largely unchanged.

Comment (PeeterPaal) [reply javascripts/editmode.js]: Took a closer look – this part seems to be a bit unnecessary, as crop toggles are not present in every design. This logic also seems tied to product & article images, which went unnoticed.

I'd keep other general scripts (bgpicker, custom data etc.), as these are perhaps more widely used.

File: .prettierrc [hunk 22]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (t-p-developer) [inline .prettierrc:1]: 🏅

File: .prettierrc [hunk 23]
```diff
@@ -0,0 +1,8 @@
...
+  "arrowParens": "avoid",
+  "bracketSameLine": false,
+  "bracketSpacing": false,
+  "printWidth": 100,
+  "singleQuote": true,
+  "trailingComma": "es5"
+}
```
Comment (t-p-developer) [inline .prettierrc:8]: 🏅

**Review by tanelj — APPROVED** [review]
Nice!

File: components/bg-picker-scripts.tpl [hunk 24]
```diff
@@ -0,0 +1,30 @@
+<script type="text/javascript">
+  $(function() {
+    $('.bg-picker-area').each(function(index, pickerArea) {
+      var $picker = $(pickerArea).find('.bg-picker');
```
Comment (tanelj) [inline components/bg-picker-scripts.tpl:4]: Its nice to guard against missing `bg-picker` element.

File: components/template-styles.tpl [hunk 25]
```diff
@@ -0,0 +1,12 @@
+{% comment %}Body background styles{% endcomment %}
+<style>
+  .body-bg-color {
+    background-color: {{ body_bg_color }};
+  }
+
+  .body-bg-image {
+    background-image: url("{{ body_bg_image }}");
```
Comment (tanelj) [inline components/template-styles.tpl]: NB! Do not render when `body_bg_image` is empty (renders current page html when url is missing).

```suggestion
    {% if  body_bg_image != blank%}background-image: url("{{ body_bg_image }}");{% endif %}

File: components/header.tpl [hunk 26]
```diff
@@ -0,0 +1,11 @@
+<div class="header">
+  {% include "menu-main" %}
+
+  <div class="header-right">
+    {% include "menu-lang" %}
+
+    {% if site.search.enabled %}
```
Comment (tanelj) [inline components/header.tpl]:
```suggestion
    {%- if site.search.enabled %}
```

File: components/header.tpl [hunk 27]
```diff
@@ -0,0 +1,11 @@
...
+      {% include "search" %}
+    {% endif %}
```
Comment (tanelj) [inline components/header.tpl]:
```suggestion
    {%- endif %}
```

File: components/javascripts.tpl [hunk 28]
```diff
@@ -0,0 +1,36 @@
...
+{% comment %}Site search related javascript components.{% endcomment %}
+{% if site.search.enabled -%}
+  <script src="{{ site.static_asset_host }}/libs/edicy-search/latest/edicy-search.js"></script>
+  <script>site.bindSiteSearch($('.js-search-form').get(0), '{{ page.language_code }}', {{ "search_noresults" | lc | json }});</script>
+{%- endif %}
+
+{% comment %}SITE ANALYTICS INITIATION{% endcomment %}
+{% unless editmode %}{{ site.analytics }}{% endunless %}
+
+{% if editmode %}
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
{%- if editmode %}
```

File: components/javascripts.tpl [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+
+      var pageData = new Edicy.CustomData({
+        type: 'page',
+        id: '{{ page.id }}'
+      });
+
+    </script>
+    {%- include "bg-picker-scripts" -%}
+  {% endeditorjsblock %}
+{% endif %}
```
Comment (tanelj) [inline components/javascripts.tpl]:
```suggestion
{%- endif %}
```

File: components/template-meta.tpl [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+<meta property="og:site_name" content="{{ page.site_title | escape_once }}">
+
+{%- comment -%}Open Graph image{%- endcomment -%}
+{%- if og_obj.og_image? -%}
+  {%- assign og_image = og_obj.og_image.for-width-1200 -%}
+{%- endif -%}
+
+{%- if og_image -%}
+  {%- comment -%}"http:" and "https:" strings are removed and readded to ensure that older bg-picker images will have protocol.{%- endcomment -%}
+  {%- if og_image.url -%}<meta property="og:image" content="{{ og_image.url | replace_first: "http:", "" | replace_first: "https:", "" | prepend: "https:" }}">{%- endif -%}
```
Comment (tanelj) [inline components/template-meta.tpl]: The `og_image` is always Image object (is not coming from object data attribute) so:
```suggestion
  {{% if og_image.url %}<meta property="og:image" content="{{ og_image.url }}">{% endif %}
```

File: layouts/front_page.tpl [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+<html class="{% if editmode %}editmode{% else %}publicmode{% endif %}" lang="{{ page.language_code }}">
+  <head prefix="og: http://ogp.me/ns#">
+    {%- include "html-head" -%}
+    {%- include "template-styles" -%}
+  </head>
+  <body class="bg-picker-area body-bg-picker-area js-background-type {{ body_bg_type }}">
+    <div class="body-bg-image js-background-image"></div>
+    <div class="body-bg-color js-background-color"></div>
+
+    {% if editmode %}
```
Comment (tanelj) [inline layouts/front_page.tpl]:
```suggestion
    {%- if editmode %}
```

File: layouts/front_page.tpl [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+        data-type_picture="true"
+        data-type_color="true"
+        data-color_elem=".body-bg-color"
+        data-picker_area_elem=".body-bg-picker-area"
+        data-picker_elem=".body-bg-picker"
+        data-bg-image="{{ body_bg_image }}"
+        data-bg-color="{{ body_bg_color }}"
+        data-bg-image-sizes="{{ body_bg_image_sizes_str | escape }}"
+      ></button>
+    {% endif %}
```
Comment (tanelj) [inline layouts/front_page.tpl]:
```suggestion
    {% endif -%}
```

File: manifest.json [hunk 33]
```diff
@@ -0,0 +1,144 @@
...
+      "file": "components/template-variables.tpl",
+      "layout_name": "template-variables",
+      "title": "template-variables"
+    },
+    {
+      "content_type": "page",
+      "component": false,
+      "file": "layouts/front_page.tpl",
+      "layout_name": "page_default",
+      "title": "Front page"
```
Comment (tanelj) [inline manifest.json]: Maybe we should name it common page?

```suggestion
      "file": "layouts/common_page.tpl",
      "layout_name": "page_default",
      "title": "Common page"
```

**Review by tanelj — CHANGES_REQUESTED** [review]
Nice work! Here are some feedback.

**Review by tanelj — APPROVED** [review]
Nice work!

**Pull request merged**